### ISSUE TITLE
Workspace browser in vim split

### ISSUE THREAD

**Issue by tdy** [daeyun/vim-matlab#14]
It'd be nice to have a workspace summary in a read-only vim split (like a `topright new` split with `buftype=nofile bufhidden=wipe nobuflisted noswapfile nowrap nomodifiable`).

As for how to keep this vim buffer in sync with MATLAB's workspace, I'm not sure.

**1\)** My first thought was to use `whos`, which gives a condensed version of the GUI workspace browser (without min/max/range/mean/median/mode/std/var):

```
>> whos
  Name     Size     Bytes     Class     Attributes
  S        4x1       6230     struct
  ans      5x5        200     double
  p       64x3       1536     double
  x        1x1          8     double
```

Use `MatlabCliController#run_code` to store `whos` output in a new `diary` after each command.

```python
code += ",delete '/tmp/vim-matlab.whos',diary '/tmp/vim-matlab.whos',whos,diary off"
```

And create `PythonVimUtils#get_current_matlab_workspace` to update `vim.current.buffer[0]` with the contents of /tmp/vim-matlab.whos (or maybe just open the file itself).

But this would output `whos` to the console after each command. As far as I can tell, `pexpect`'s filters wouldn't help in this case.

**2\)** My second thought... I don't have one yet.

**Comment by daeyun** [comment]
How about using MATLAB's GUI window as-is (`workspace` command in cli)?

Similarly you can do `edit {filename}` or (`:MatlabCliOpenInEditor` in vim) to open the GUI editor for things like setting breakpoints.

<img width="512" alt="screenshot 2016-08-18 20 57 56" src="https://cloud.githubusercontent.com/assets/1250682/17795452/82df34fe-6586-11e6-8049-6fc6712c9b0e.png">

**Comment by tdy** [comment]
That's true, but I was just thinking about potential functionalities that could be moved to vim/multiplexers without the JVM.

This is just a lower priority feature request in case someone has the time/interest/idea for implementing it. Actually I don't know how to submit feature requests on github, so I just used Issues. Not sure if that's how it works on here.

**Comment by daeyun** [comment]
Yeah sure. I think it would be useful if someone can implement it. This is a hobbyist project anyway; any patches or requests are welcome any time. I really appreciate it.

I agree that `diary` is the way to go about this. For preventing spamming the console every time, something that can temporarily turn on and off the console output would be useful in general, for other features too. Probably implemented as having a marker string to indicate where to start and stop ignoring. I think a related issue is command history; currently when you press ⬆️ arrow key, all the helper code shows up, and if we were to use a marker, it would be useful to have a way to hide that from command history too.
